lib/watir/elements/select.rb

```diff
@@ -29,9 +29,16 @@ def include?(str_or_rx)
     # @return [String] The text of the option selected. If multiple options match, returns the first match.
     #
 
-    def select(*str_or_rx)
-      results = str_or_rx.flatten.map { |v| select_by v }
-      results.first
+    def select(*both, text: nil, value: nil, label: nil)
+      selection = {both: both, text: text, value: value, label: label}.select do |_k, v|
+        !v.nil? && ![v].flatten.empty?
+      end
+      raise 'Can not select by more than one method' if selection.size > 1
+
+      value = selection.values.first
+      type_check(value)
+
+      [value].flatten.map { |v| select_by v }.first
```

Comment:
I like the idea of being able to specify how to locate the option. I know I've had at least one time where this would have helped.

However... we don't actually seem to be doing anything with it - ie shouldn't selection.keys.first get based along/used?

Reply:
Yeah, optional keywords for the win again.
`#first` is returning the value of the first item selected. Not certain why that's what we want returned, but that's the backward compatible thing.

Reply:
I meant that `#select_by` does not know to select by `:text`, `:value`, vs `:label`. The specified keyword gets ignored.

For example, given the page:

~~~~~~~~
<html>
  <body>
    <select>
      <option value="a">1</option>
      <option value="b">2</option>
      <option value="c">3</option>
    <select>
  </body>
</html>
~~~~~~~~

If I try to to pick the `:value` of "3":

~~~~~~~~
browser.select_list.select(value: '3')
p browser.select_list.selected_options.map(&:text)
#=> ["3"]
~~~~~~~~

The option with `:text` of "3" has been selected (unexpected). I would only expect to pick options with the matching value attribute.